section_faiss_vector_index.go

```diff
@@ -255,6 +255,26 @@ func (v *vectorIndexOpaque) flushVectorSection(vecToDocID map[int64]*roaring.Bit
 	return fieldStart, nil
 }
 
+// Func which returns if reconstruction is required.
+// Won't be required if multiple flat indexes are being combined into a larger
+// flat index.
+func reconstructionRequired(isNewIndexIVF bool, indexes []vecIndexMeta) bool {
+	// Always required for IVF indexes.
+	if isNewIndexIVF {
+		return true
+	}
+
+	// if any existing index is not flat, all need to be reconstructed.
+	for _, index := range indexes {
+		_, isExistingIndexIVF := getIndexType(len(index.vecIds) + len(index.deleted))
```

Comment:
TODO Food for thought: Would this cause an issue with forward compatibility if the limits deciding flat/IVF are changed in a newer zapx version the user upgrades to?